src/main/cpp/src/parse_uri.cu

```diff
@@ -951,36 +954,85 @@ std::unique_ptr<column> parse_uri(strings_column_view const& input,
                              std::move(null_mask));
 }
 
+void validate_output(strings_column_view const& input,
+                     column const& result,
+                     rmm::cuda_stream_view stream,
+                     rmm::device_async_resource_ref mr)
+{
+  if (input.size() == 0) { return; }
+
+  // In ANSI mode, check for completely invalid URIs
+  auto const d_strings = column_device_view::create(input.parent(), stream);
+
+  // Create boolean vector: true for valid URIs, false for completely invalid URIs
+  auto validity_flags = rmm::device_uvector<bool>(input.size(), stream);
+
+  thrust::tabulate(
+    rmm::exec_policy_nosync(stream),
+    validity_flags.begin(),
+    validity_flags.end(),
+    cuda::proclaim_return_type<bool>([input = *d_strings] __device__(cudf::size_type row_idx) {
+      // Input nulls remain null
+      if (input.is_null(row_idx)) { return false; }
+
+      auto const in_string = input.element<string_view>(row_idx);
+      auto const uri =
+        validate_uri(in_string.data(), in_string.size_bytes(), cuda::std::nullopt, row_idx);
+
+      return uri.valid != 0;
+    }));
+
+  auto [validation_mask, null_count] = cudf::detail::valid_if(
+    validity_flags.begin(), validity_flags.end(), cuda::std::identity{}, stream, mr);
+
+  // Create validation column for throw_row_error_if_any
+  auto validation_column = std::make_unique<column>(
+    std::move(validity_flags),
+    null_count > 0 ? std::move(validation_mask) : rmm::device_buffer{0, stream},
+    null_count);
+  throw_row_error_if_any(input.parent(), validation_column->view(), stream);
+}
+
 }  // namespace detail
 
 // external API
 
 std::unique_ptr<column> parse_uri_to_protocol(strings_column_view const& input,
+                                              bool ansi_mode,
                                               rmm::cuda_stream_view stream,
                                               rmm::device_async_resource_ref mr)
 {
   CUDF_FUNC_RANGE();
-  return detail::parse_uri(input, detail::URI_chunks::PROTOCOL, std::nullopt, stream, mr);
+  auto output = detail::parse_uri(input, detail::URI_chunks::PROTOCOL, std::nullopt, stream, mr);
+  if (ansi_mode) { detail::validate_output(input, *output, stream, mr); }
```

Comment:
This does not return any result thus we don't need to pass `mr`. The validation function only uses the default memory resource.